Retranslate allocate from 分配 to 配置

Risheng1128 opened the pull request:
Allocate is translated as 配置 rather than 分配 in Tainwan based on [大陸台灣計算機術語對照表.
](https://zh.wikibooks.org/zh-tw/%E5%A4%A7%E9%99%86%E5%8F%B0%E6%B9%BE%E8%AE%A1%E7%AE%97%E6%9C%BA%E6%9C%AF%E8%AF%AD%E5%AF%B9%E7%85%A7%E8%A1%A8) Meanwhile, allocator is called 配置器.

weihanglo commented on src/ch20-02-multithreaded.md:
```diff
@@ -26,7 +26,7 @@
 ### 透過執行緒池改善吞吐量
 
-**執行緒池**（thread pool）會產生一群執行緒來等待並隨時準備好處理任務。當程式收到新任務時，它會將此任務分配給執行緒池其中一條執行緒，然後該執行緒就會處理該任務。池中剩餘的執行緒在第一條執行緒處理任務時，仍能隨時處理任何其他來臨的任務。當第一條執行緒處理完成時，他會回到閒置執行緒池之中，等待處理新的任務。執行緒池讓你能並行處理連線，增加伺服器的吞吐量。
+**執行緒池**（thread pool）會產生一群執行緒來等待並隨時準備好處理任務。當程式收到新任務時，它會將此任務配置給執行緒池其中一條執行緒，然後該執行緒就會處理該任務。池中剩餘的執行緒在第一條執行緒處理任務時，仍能隨時處理任何其他來臨的任務。當第一條執行緒處理完成時，他會回到閒置執行緒池之中，等待處理新的任務。執行緒池讓你能並行處理連線，增加伺服器的吞吐量。
```
> …it assigns one of the threads in the pool to the task, and that thread will process the task

This is assign not allocate

Risheng1128 replied:
Thanks. I think 指派 is better than 配置 and 分配.

weihanglo commented on src/ch12-03-improving-error-handling-and-modularity.md:
```diff
@@ -12,7 +12,7 @@
 ### 分開執行檔專案的任務
 
-`main` 函式負責多數任務的組織分配問題在許多執行檔專案中都很常見。所以 Rust 社群開發出了一種流程，這在當 `main` 開始變大時，能作為分開執行檔程式中任務的指導原則。此流程有以下步驟：
+`main` 函式負責多數任務的組織配置問題在許多執行檔專案中都很常見。所以 Rust 社群開發出了一種流程，這在當 `main` 開始變大時，能作為分開執行檔程式中任務的指導原則。此流程有以下步驟：
```
> The organizational problem of allocating responsibility for multiple tasks to the main function is common to many binary projects. 

Although it does use the verb “allocating”, it is not necessary to change since it is more like “assigning task” not “allocating memory”. Up to you.

Risheng1128 replied:
Ditto, I think 指派 is better.

weihanglo commented on src/ch16-03-shared-state.md:
````diff
@@ -113,7 +113,7 @@ changes in the compiler -->
 ```
 
-我們辦到了！我們從 0 數到了 10，雖然看起來不是很令人印象深刻，但這的確教會了我們很多有關 `Mutex<T>` 與執行緒安全的知識。你也可以使用此程式結構來做更多複雜的運算，而不只是數數而已。使用此策略，你可以將運算拆成數個獨立部分，將它們分配給執行緒，然後使用 `Mutex<T>` 來讓每個執行緒更新該部分的結果。
+我們辦到了！我們從 0 數到了 10，雖然看起來不是很令人印象深刻，但這的確教會了我們很多有關 `Mutex<T>` 與執行緒安全的知識。你也可以使用此程式結構來做更多複雜的運算，而不只是數數而已。使用此策略，你可以將運算拆成數個獨立部分，將它們配置給執行緒，然後使用 `Mutex<T>` 來讓每個執行緒更新該部分的結果。
````
> …you can divide a calculation into independent parts, split those parts across threads…

This is “split” not allocate.

Risheng1128 replied:
I think it can be retained.

weihanglo commented on src/ch20-01-single-threaded.md:
```diff
@@ -232,7 +232,7 @@ HTTP/1.1 200 OK\r\n\r\n
 ### 再做一些重構
 
-目前 `if` 與 `else` 區塊有很多重複的地方，它們都會讀取檔案並將檔案內容寫入流中。唯一不同的地方在於狀態行與檔案名稱。讓我們將程式碼變得更簡潔，將不同之處分配給 `if` 與 `else`，它們會分別將相對應的狀態行與檔案名稱賦值給變數。我們就能使用這些變數無條件地讀取檔案並寫入回應。範例 20-9 顯示了替換大段 `if` 與 `else` 區塊後的程式碼。
+目前 `if` 與 `else` 區塊有很多重複的地方，它們都會讀取檔案並將檔案內容寫入流中。唯一不同的地方在於狀態行與檔案名稱。讓我們將程式碼變得更簡潔，將不同之處配置給 `if` 與 `else`，它們會分別將相對應的狀態行與檔案名稱賦值給變數。我們就能使用這些變數無條件地讀取檔案並寫入回應。範例 20-9 顯示了替換大段 `if` 與 `else` 區塊後的程式碼。
```
> …by pulling out those differences into separate if and else lines

This is not allocation.

Risheng1128 replied:
Ditto, retain the original translation.

weihanglo commented on src/ch17-03-oo-design-patterns.md:
```diff
@@ -240,7 +240,7 @@
 ## 總結
 
-無論你讀完此章後，認為 Rust 是否屬於物件導向語言，你都知道在 Rust 中你可以使用特徵物件來取得一些物件導向的特色。動態分配能給予你的程式碼更多的彈性，但會犧牲一點執行時效能。你可以使用此彈性來實作物件導向模式，幫助提升程式碼可維護性。Rust 還有其他像是所有權等物件導向語言所沒有的功能。物件導向模式不會永遠是善用 Rust 潛能的最佳方案，不過仍是個不錯的選項。
+無論你讀完此章後，認為 Rust 是否屬於物件導向語言，你都知道在 Rust 中你可以使用特徵物件來取得一些物件導向的特色。動態配置能給予你的程式碼更多的彈性，但會犧牲一點執行時效能。你可以使用此彈性來實作物件導向模式，幫助提升程式碼可維護性。Rust 還有其他像是所有權等物件導向語言所沒有的功能。物件導向模式不會永遠是善用 Rust 潛能的最佳方案，不過仍是個不錯的選項。
```
> … Dynamic dispatch can give your code some flexibility in exchange for a bit of runtime performance.

This is dynamic dispatch not dynamic allocation.

Risheng1128 replied:
I think dynamic dispatch can be called 動態分派 based on [動態分派](https://zh.wikipedia.org/zh-hant/%E5%8A%A8%E6%80%81%E5%88%86%E6%B4%BE)

weihanglo reviewed (changes requested):
Thanks for all the fixes.

Some places do not really refer to “memory allocation”. Please have a look at my suggestions :)

CYBAI reviewed (commented):
How about let's focus on updating translation for `allocate` in this PR and leave others in separate issues?

Risheng1128 committed:
Retranslate allocate from 分配 to 配置

Allocate is translated as 配置 rather than 分配 in Taiwan.
Meanwhile, allocator is called 配置器.

Risheng1128 commented:
TODO:
1. assign -> 指派
2. dynamic dispatch -> 動態分派

weihanglo reviewed (approved):
Thanks a lot!

weihanglo commented:
@bors r+

weihanglo commented:
oh this is not a bors repo. No `r+` support 😅.
